src/Meshes/mesh.ts

```diff
@@ -3870,8 +3898,12 @@ export class Mesh extends AbstractMesh implements IGetSetVerticesData {
         }
 
         // Material
-        if (parsedMesh.materialId) {
-            mesh.setMaterialById(parsedMesh.materialId);
+        const materialUniqueId = parsedMesh.materialUniqueId;
+        const materialId = parsedMesh.materialId;
+        if (materialUniqueId) {
+            mesh.setMaterialByUniqueId(materialUniqueId);

```

Comment:
This is where I disagree. Because if 2 scenes comes with the same uniqueId, this system will not work. What we need to do is to store the materialArray (the one containing the key (uniqueId)-> material) at a higher level so the mesh loader can use it to retrieve the current (for this file) mapping of uniqueIds

I hope it makes sense :)

Reply:
Look what I did with tempIndexContainer in the babylon file loader